ERCS/erc-7722.md

```diff
@@ -0,0 +1,328 @@
+---
+eip: 7722
+title: Opaque Token
+description: A token specification designed to enhance privacy by concealing balance information.
+author: Ivica Aračić (@ivica7), SWIAT
+discussions-to: https://ethereum-magicians.org/t/erc-7722-opaque-token/20249
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-06-09
+---
+
+## Abstract
+
+This ERC proposes a specification for an opaque token that enhances privacy by concealing balance information. Privacy is achieved by representing balances as off-chain data encapsulated in hashes, referred to as "baskets". These baskets can be reorganized, transferred, and managed through token functions on-chain.
+
+## Motivation
+
+In [ERC-20](./eip-20.html), token balances are transparent on-chain. While some privacy can be achieved by using pseudonyms for the sender and receiver, the risk of revealing the entire portfolio to the public increases when using smart contract accounts like Gnosis Safe or [ERC-725](./eip-725.html). This is particularly problematic in regulated environments where KYC (Know Your Customer) and KYBP (Know Your Business Partner) procedures are required.
+
+This proposal aims to conceal balances on-chain, allowing the use of smart contract accounts to hold tokens without compromising privacy or integrity.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
```

Comment:
This section could do with an overview of the entire system. It's a bit hard to grok just from the interfaces and endpoints.

Reply:
added a summary about the core concept around which this token proposal revolves around.